app/src/main/java/com/github/doomsdayrs/apps/shosetsu/backend/TarArchive.kt

```diff
@@ -0,0 +1,75 @@
+package com.github.doomsdayrs.apps.shosetsu.backend
+
+import org.kamranzafar.jtar.*
+import java.io.File
+import java.io.InputStream
+import java.io.RandomAccessFile
+
+class TarArchive(val file: File) {
+
+
+    fun readTar(name: String): InputStream {
+        val tar = TarInputStream(file.inputStream())
+        var entry = tar.nextEntry
+        var buffer = ByteArray(0)
+        while (entry != null) {
```

Comment:
See if you can use an iterator based loop instead of a while loop.

Something like:

```
for (entry in tar) {
    // Do something
}
```

Or better yet if you can get it as a collection by some pre-processing,
```
files.find( file -> file.name.contains(name, true))
```

This is just me giving you ideas, I haven't looked at the code with proper perspective so you don't have to make all these changes.

Reply:
Theres no way to know number of entries before iterating manually in tar. At least not on the library(Jtar) i am using.

Reply:
you don't need to know the number of entries, that's the magic of iterators.

You would have to write the iterator though.

Reply:
I think current way might be better because now it only gets entries until it find what it need. But if i make iterator it will continue until all entries are read even after going over what it was searching for.

Reply:
The tar format was designed without a centralized index or table of content for files and their properties for streaming to tape backup devices. The archive must be read sequentially to list or extract files.

Reply:
Could you give an example. The iterator  i find by searching google needs list variable?

Reply:
replaced while loop with sequence.